TechnicalDocuments/Proposals/BabylonDesignLibrary.md

```diff
@@ -0,0 +1,196 @@
+        let secondary: UIColor
+    }
+}
+
+extension DesignLibrary {
+    struct ComponetsBuilder: BoxAppearance {
+        let configuration: Configuration
+        var traits: UITraitCollection
+
+        @available(*, deprecated, message: "Use DesignLibrary.Configuration")
+        public var brandColors: BrandColorsProtocol = DefaultBrandColor()
+
+        @available(*, deprecated, message: "Use DesignLibrary.Configuration")
+        public var appColors: AppColors = AppColors()
+
+        @available(*, deprecated, message: "Use DesignLibrary.componentBuilder")
+        public var modern: ModernPalette {
+            return ModernPalette(self)
+        }
+
+        fileprivate init(configuration: Configuration, traits: UITraitCollection) {
+            self.configuration = configuration
+            self.traits = traits
+        }
+
+        func primaryButton(title: String, isEnabled: Bool, didTap: (() -> Void)?) -> AnyRenderable { ... }
+
+        func secondaryButton(title: String, isEnabled: Bool, didTap: (() -> Void)?) -> AnyRenderable { ... }
+
+        func loader() -> AnyRenderable { ... }
+    }
+}
+
+extension DesignLibraryService {
+    struct ViewControllerBuilder {
+        let configuration: Configuration
+
+        func makeConfirmationModal(
+            title: String,
+            description: String,
+            acceptButtonTitle: String,
+            declineButtonTitle: String
+        ) -> UIViewController { ... }
+    }
+}
+```
+
+## Impact on existing codebase
+`DesignLibrary` will be accessible via `Current` and `BabylonBoxViewController` will be modified in order to
+make the change less distruptive.
+Here is the implementation of the above statement
+
+```swift
+/// extend `World`
+import Bento
+import BentoKit
+import ReactiveSwift
+
+protocol DesignLibraryProtocol {
+    associatedtype ComponentsBuilder: BoxAppearance
+    var componentsBuilder: ComponentsBuilder { get }
+}
+
+extension DesignLibrary.ComponetsBuilder: BoxAppearance {}
+extension DesignLibrary: DesignLibraryProtocol {}
+
+/// Note: DesignLibrary can't be used directly here due to
+/// framework dependencies.
+struct World<DesignLibrary: DesignLibraryProtocol> {
+    let designLibrary: MutableProperty<DesignLibrary>
+}
+
+/// Use Current in BabylonBoxViewController
+class BabylonBoxViewController<ViewModel, Renderer>: BoxViewController<
+    ViewModel,
+    Renderer,
+    DesignLibrary.ComponetsBuilder
+> where
+    ViewModel: BoxViewModel,
+    Renderer: BoxRenderer,
+    ViewModel.State == Renderer.State,
+    ViewModel.Action == Renderer.Action,
+    Renderer.DesignLibrary == DesignLibrary.ComponetsBuilder {
+    init(
+        viewModel: ViewModel,
+        renderer: Renderer.Type,
+        rendererConfig: Renderer.Config
+    ) {
+        super.init(
+            viewModel: viewModel,
+            renderer: renderer,
+            rendererConfig: rendererConfig,
+            appearance: Current.DesignLibrary.map(\.componentsBuilder)
```

Comment:
Passing this as a parameter with `Current.designLibrary` as default value would have the same effect but will allow more flexibility in case we need to replace `appearance` with something fake keeping everything else in `BabylonBoxViewController`.

Reply:
👍